File: contests/Google CodeJam - Qualification Round 2022/Round 1B 2022/b.py
def read():
    nums = input().split()
    return [int(num) for num in nums]

def solve(N, P, X):
    ans = 0
    pressure = 0
    for i in range(N):
      
        X[i].sort()

        if X[i][0] < pressure < X[i][-1]:
            if pressure - X[i][0] <= X[i][-1] - pressure:
                ans += (pressure - X[i][0]) + (X[i][-1] - X[i][0])
                pressure = X[i][-1]
            else:
                ans += (X[i][-1] - pressure) + (X[i][-1] - X[i][0])
                pressure = X[i][0]
        else:
            if pressure < X[i][-1]:
                ans += X[i][-1] - pressure
                pressure = X[i][-1]
            else:
                ans += pressure - X[i][0]
                pressure = X[i][0]
    return ans

def main():
    T = int(input())
    for case in range(1, T+1):
        N, P = read()
        X = []
        for _ in range(N):
            X.append(read())
        ans = solve(N, P, X)
        print(f"Case #{case}: {ans}")

main()

File: contests/Google CodeJam - Qualification Round 2022/Round 1B 2022/b.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <math.h>
#define ll long long
const unsigned int M = 1000000007;
using namespace std;

ll solve(vector<vector<ll>>v, int p, int n) {
	ll ans = 0, pressure = 0;
	
	for(int i = 0; i < n; i++) {
		sort(v[i].begin(), v[i].end());

		if(pressure > v[i][0] && pressure < v[i][(n-1)]) {
			if(pressure - v[i][0] <= v[i][(n-1)] - pressure) {
				ans += (pressure - v[i][0]) + (v[i][(n-1)] - v[i][0]) % M;
				pressure = v[i][(n-1)];
			} else {
				ans += (v[i][(n-1)] - pressure) + (v[i][(n-1)] - v[i][0]) % M;
				pressure = v[i][0];
			}
		} else {
			if(pressure < v[i][(n-1)]) {
				ans += (v[i][(n-1)] - pressure)  % M;
				pressure = v[i][(n-1)];
			} else {
				ans += (pressure - v[i][0]) % M;
				pressure = v[i][0];
			}
		}
	}

	return ans;
}

int main() {
	int T, test = 1;
	cin>>T;
	while(T--) {
		int N, P;
		cin>>N>>P;
		vector<vector<ll>>X(N);
		for(int i = 0; i < N; i++) {
			X[i] = vector<ll>(P);
			for(int j = 0; j < P; j++) {
				cin>>X[i][j];
			}
		}
		// cout<<"X"<<endl;
		// for(int i = 0; i < N; i++) {
		// 	for(int x : X[i]) cout<<x<<" ";
		// 	cout<<endl;
		// }
		cout<<"Case #"<<test<<": "<<solve(X, P, N)<<endl;
		test++;
	}
	return 0;
}